### PR TITLE
Filter unsupported relation for RangeQueryBuilder

### LINKED ISSUE #26575: Range queries on range fields accept `relation: disjoint` but treat it as `relation: intersects`
This can be confusing since `disjoint` means the opposite of `intersects`.

### PR DESCRIPTION
**Pull request by liketic**
Closes #26575
@jpountz Please help to review. Happy to make further changes. Thanks.

### CODE REVIEW

**Review by jpountz — CHANGES_REQUESTED** [review]
I left a comment. It looks good to me in general, thanks for writing a test!

**Comment by liketic** [comment]
@jpountz Thanks! However I didn't find your comment...  😃

File: core/src/main/java/org/elasticsearch/common/geo/ShapeRelation.java [hunk 1]
```diff
@@ -65,4 +65,8 @@ public static ShapeRelation getRelationByName(String name) {
     public String getRelationName() {
         return relationName;
     }
+
+    public boolean isAllowedInRangeQuery() {
+        return this == INTERSECTS || this == CONTAINS || this == WITHIN;
+    }
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/common/geo/ShapeRelation.java]: It feels a bit wrong to me to make ShapeRelation aware of how it is used in range queries. I'd rather put this logic in RangeQueryBuilder.

**Comment by jpountz** [comment]
I had forgotten to click on the button. :)

File: core/src/main/java/org/elasticsearch/common/geo/ShapeRelation.java (hunk 1 above)
Comment (liketic) [reply core/src/main/java/org/elasticsearch/common/geo/ShapeRelation.java]: Make sense! Very thanks!

**Review by jpountz — APPROVED** [review]

**Comment by jpountz** [comment]
@elasticmachine please test it

**Comment by liketic** [comment]
@jpountz Please let me know if you plan to merge this so I can resolve the conflicts and squash commits. Thanks very much! 😄

**Comment by jpountz** [comment]
CI check was successful so I'll merge it today.

**Comment by liketic** [comment]
@jpountz rebase done. Thanks!

**Pull request merged**